gateway_test.go

```diff
@@ -991,3 +992,46 @@ func TestListenPathTykPrefix(t *testing.T) {
 	}
 	testHttp(t, tests, false)
 }
+
+func TestProxyUserAgent(t *testing.T) {
+	spec := createSpecTest(t, sampleAPI)
+	remote, _ := url.Parse(spec.Proxy.TargetURL)
+	proxy := TykNewSingleHostReverseProxy(remote, spec)
+	proxyHandler := ProxyHandler(proxy, spec)
+
+	tests := []struct {
+		sent   interface{}
+		wantRe string
+	}{
+		// none set; use our default
+		{nil, `^Tyk/v\d+\.\d+\.\d+$`},
+		// set but empty; let it through
+		{"", `^$`},
+		// set and not empty
+		{"SomeAgent", `^SomeAgent$`},
+	}
+	for _, tc := range tests {
+		rec := httptest.NewRecorder()
+		req := testReq(t, "GET", "/sample", nil)
+		if s, ok := tc.sent.(string); ok {
+			req.Header.Set("User-Agent", s)
+		}
```

Comment:
If anyone wonders, I tried using `*string` but it made the table-based tests impossible - you can't do `&"foo"`.